modules/andes-core/broker/src/main/java/org/dna/mqtt/moquette/messaging/spi/impl/ProtocolProcessor.java

```diff
@@ -688,19 +757,21 @@ void processSubscribe(ServerChannel session, SubscribeMessage msg, String client
             }
         }
 
-        //ack the client
-        SubAckMessage ackMessage = new SubAckMessage();
-        ackMessage.setMessageID(msg.getMessageID());
+        // Ack the client only if all subscriptions are successful
+        if (authenticated) {
```

Comment:
If we send success, the subscription which was failed also will think that it was successful. Not sure which is the correct way?